src/scripts/populate-dev.js

```diff
@@ -0,0 +1,135 @@
+    // find duplicate
+    let clean_data = _.omit(data, _.keys(mapping))
+    // console.log("clean_data",clean_data)
+    let existing = await klass.findOne(clean_data)
+
+    if (!existing) {
+
+        let find_result = {}
+        let mapped_data = _.pick(data, _.keys(mapping))
+        let keys = _.keys(mapped_data)
+        let find_err = false
+
+        for (let k of keys) {
+            let res = await mapping[k](data[k])
+
+            if (res) {
+                //collect find_result
+                find_result = Object.assign(find_result, res)
+            }
+            else {
+                find_err = true
+                console.log(`mapping error: ${k} not found`, data[k])
+                break
+            }
+        }
+
+        if (!find_err) {
+            // merge custom mapped field back to data
+            let new_data = Object.assign(clean_data, find_result)
+            let row = new klass(new_data)
+            await row.save()
+            console.log("saving row", data)
+        }
+    }
+    else {
+        console.log("duplicate row", data)
+    }
+}
+async function parse_json(tableName, content) {
+    // json content file has to be an array
+    const fileContent = JSON.parse(content);
+    let klass = getKlass(tableName)
+    for (let row of fileContent) {
+        try {
+            await saveRow(klass, row)
+        } catch (err) {
+            console.log('save exception ' + err.message, row)
+        }
+
+    }
+}
+
+async function readJson(t, filepath) {
+    try {
+        console.log('start file ', filepath)
+        let content = await FS.read(filepath)
+        await parse_json(t, content)
+        console.log('file processing finished', filepath, "\n\n")
+    } catch (err) {
+        console.log('readJson error ' + err)
+    }
+}
+
+function init() {
+    validateEnvVariables();
+    mongooseConf(mongoose);
+}
+
+
+/**
+ * convert map-user to user.id
+ */
+let user_id_functor = async (data) => {
+    let klass = getKlass('user')
+    let user = await klass.findOne(data)
+    if (user)
+        return {user: user._id}
+    return null
+}
+
+/**
+ * convert map-local to local by hashing local.password
+ */
+let password_functor = async (data) => {
+    let ret = data
+    let klass = getKlass('user')
+
+    ret.password = new klass().generateHash(ret.password)
+    return {local:ret}
+}
+/**
+ * mapping dictionary that map a field in the dev-data jsons
+ */
+const mapping = {
+    "map-user": user_id_functor,
+    "map-local": password_functor
+}
+
+/**
+ * this is es7 code so has to be called by a shell that has babel-register,e.g populate-runner.js
+ * @param tables are file name in dev-data folder that will be imported to the database
+ * @returns {Promise.<void>}
+ */
+module.exports = async function start(tables) {
+
+    init()
+    //  await test() // when developing new mapping functor
+    for (let t of tables) {
+        await readJson(t, `./dev-data/${t}.json`)
+    }
+    /*
```

Comment:
I fixed the file as requested.

klass is renamed to model. It is a way to programmaticly import js module, eg 'user.model.js'